src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1955,8 +1955,18 @@ private static Expression BindParameterFromExpression(
                 Expression.Convert(Expression.Constant(parameter.DefaultValue), parameter.ParameterType)));
     }
 
-    private static Expression BindParameterFromProperty(ParameterInfo parameter, MemberExpression property, PropertyInfo itemProperty, string key, RequestDelegateFactoryContext factoryContext, string source) =>
-        BindParameterFromValue(parameter, GetValueFromProperty(property, itemProperty, key, GetExpressionType(parameter.ParameterType)), factoryContext, source);
+    private static Expression BindParameterFromProperty(ParameterInfo parameter, MemberExpression property, PropertyInfo itemProperty, string key, RequestDelegateFactoryContext factoryContext, string source)
+    {
+        var expressionType = GetExpressionType(parameter.ParameterType);
```

Comment:
Are string arrays the only type that we need to handle here? Does MVC support serializing `int[]` from the header for example?

Reply:
@captainsafia 

When examining `var expressionType` I am bearing in mind that only `typeof(string[])` will return from `GetExpressionType(parameter.ParameterType)` for any array. 

In other words, the line
`var valueExpression = (source == "header" && expressionType == typeof(string[]))`

Is equivalent to

`var valueExpression = (source == "header" && parameter.ParameterType.IsArray)`


I altered the code to the second version.